### PR TITLE
remove nulls when updating documents

### LINKED ISSUE #396: UpdateDocumentsAsync replacing whole document in index
Hello,
following problem with method UpdateDocumentsAsync.
**Description**
When using UpdateDocumentsAsync the put method if HttpClient is used. The meilisearch docs state, that using this route/method will

> If you send an already existing document (same [document id]) the old document will be only partially updated according to the fields of the new document. Thus, any fields **not present** in the new document are **kept** and **remain unchanged**.

Using a modified example from the tests
```c#
var task = await index.AddDocumentsAsync(new[]
            {
                new Movie { Id = "1", Name = "Batman", Genre = "Action" },
                new Movie { Id = "2", Name = "Superman", Genre = "Action" },
            });

task = await index.UpdateDocumentsAsync(new[] { new Movie { Id = "1", Name = "Ironman" } });
await index.WaitForTaskAsync(task.TaskUid);
var doc = await index.GetDocumentAsync<Movie>(1);

Console.WriteLine(doc.Genre); // outputs null/empty
```

**Expected behavior**
Just a partial update of the document, when not setting all fields. 

**Current behavior**
Current behavior is more like a complete replace of the existing document in the index. In above example Genre is set to null. It seems  the JsonSerializer serializes not set fields to null/empty.

You had this issue addressed with a function to remove null values, but this is gone https://github.com/meilisearch/meilisearch-dotnet/issues/231#issue-1119599955

**Environment (please complete the following information):**
- OS: Docker
- Meilisearch version: latest
- meilisearch-dotnet version: v0.13.1

### PR DESCRIPTION
**Pull request by ahmednfwela**
# Pull Request

## Related issue
Fixes #396

## What does this PR do?
- Adds `JsonSerializerOptions` to `PutJsonCustomAsync`
- Remove nulls when updating documents

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by ahmednfwela** [comment]
cc @alallema

File: tests/Meilisearch.Tests/DocumentTests.cs [hunk 1]
```diff
@@ -275,6 +275,8 @@ public async Task BasicDocumentsUpdate()
             var docs = await index.GetDocumentsAsync<Movie>();
             var movieNames = docs.Results.Select(movie => movie.Name);
 
+            //Ensure Genre didn't get removed
```
Comment (alallema) [inline tests/Meilisearch.Tests/DocumentTests.cs]:
```suggestion
            // Ensure Genre didn't get removed
```

**Review by alallema — DISMISSED** [review]
Thanks @ahmednfwela for this PR and to contribute to Meilisearch ❤️ 
LGTM! ✨ 
(I just added a space in the comment)

**Review by alallema — APPROVED** [review]

**Comment by alallema** [comment]
bors merge

**Pull request merged**